auth/azure.go

```diff
@@ -0,0 +1,93 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/coreos/mantle/platform/api/azure"
+)
+
+type AzureEnvironment struct {
+	ActiveDirectoryEndpointURL                        string `json:"activeDirectoryEndpointUrl"`
+	ActiveDirectoryGraphAPIVersion                    string `json:"activeDirectoryGraphApiVersion"`
+	ActiveDirectoryGraphResourceID                    string `json:"activeDirectoryGraphResourceId"`
+	ActiveDirectoryResourceID                         string `json:"activeDirectoryResourceId"`
+	AzureDataLakeAnalyticsCatalogAndJobEndpointSuffix string `json:"azureDataLakeAnalyticsCatalogAndJobEndpointSuffix"`
+	AzureDataLakeStoreFileSystemEndpointSuffix        string `json:"azureDataLakeStoreFileSystemEndpointSuffix"`
+	GalleryEndpointURL                                string `json:"galleryEndpointUrl"`
+	KeyVaultDNSSuffix                                 string `json:"keyVaultDnsSuffix"`
+	ManagementEndpointURL                             string `json:"managementEndpointUrl"`
+	Name                                              string `json:"name"`
+	PortalURL                                         string `json:"portalUrl"`
+	PublishingProfileURL                              string `json:"publishingProfileUrl"`
+	ResourceManagerEndpointURL                        string `json:"resourceManagerEndpointUrl"`
+	SqlManagementEndpointURL                          string `json:"sqlManagementEndpointUrl"`
+	SqlServerHostnameSuffix                           string `json:"sqlServerHostnameSuffix"`
+	StorageEndpointSuffix                             string `json:"storageEndpointSuffix"`
+}
+
+type AzureManagementCertificate struct {
+	Cert string `json:"cert"`
+	Key  string `json:"key"`
+}
+
+type AzureSubscription struct {
+	EnvironmentName       string                     `json:"environmentName"`
+	ID                    string                     `json:"id"`
+	IsDefault             bool                       `json:"isDefault"`
+	ManagementCertificate AzureManagementCertificate `json:"managementCertificate"`
+	ManagementEndpointURL string                     `json:"managementEndpointUrl"`
+	Name                  string                     `json:"name"`
+	RegisteredProviders   []string                   `json:"registeredProviders"`
+	State                 string                     `json:"state"`
+}
+
+// AzureProfile represents a parsed Azure Profile Configuration File.
+type AzureProfile struct {
+	Environments  []AzureEnvironment  `json:"environments"`
+	Subscriptions []AzureSubscription `json:"subscriptions"`
+}
+
+func (ap *AzureProfile) AsOptions() []azure.Options {
+	var o []azure.Options
+
+	for _, sub := range ap.Subscriptions {
+		newo := azure.Options{
+			SubscriptionID:        sub.ID,
```

Comment:
A lot more of these settings are going to change. Just about everything in `AzureEnvironment` will defer depending on the region.